### ISSUE TITLE
RF05: Do not use special characters in identifiers in from clause

### ISSUE THREAD

**Issue by rwork0** [sqlfluff/sqlfluff#6718]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When the table name is quoted because of a special character, the RF05 rule fails even though the script does not have control over the table name.

Example query:
```sql
SELECT [foo].[abc]
FROM [foo$] AS [foo]

``` 

### Expected Behaviour

The rule should not fail.

### Observed Behaviour

L:   2 | P:   6 | RF05 | Do not use special characters in identifiers.
                       | [references.special_chars]

### How to reproduce

test.sql>
SELECT [foo].[abc]
FROM [foo$] AS [foo]

`sqlfluff lint test.sql --dialect tsql`

### Dialect

tsql

### Version

sqlfluff, version 3.3.1
Python 3.11.7

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
You can configure RF05 to allow special characters in names. See documentation here: https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-RF05

For example this will remove the lint error for your example:

```ini
[sqlfluff:rules:references.special_chars]
additional_allowed_characters = $
```

I'm closing this, but if you have more questions feel free to open a new issue.
